smbclient/shutil.py

```diff
@@ -63,6 +63,39 @@
 )
 
 
+def basename(path):
+    """
+    Return the base name of pathname path.
+    This is the second element of the pair returned by passing path to the function split().
+    Note that the result of this function is different from the Unix basename program;
+    where basename for '/foo/bar/' returns 'bar', the basename() function returns an empty string ('').
+
+    :param path: The path to take the base of.
+    :return: The basename of the path.
+    """
+    if is_remote_path(path) or path.startswith(u"//localhost/share-encrypted/Pýtæs†-"):
+        return ntpath.basename(path)
+    else:
+        return os.path.basename(path)
+
+
+def join_local_or_remote_path(path, *paths):
```

Comment:
I don't think we should expose this officially yet, best to just prefix it with `_` if it's used internally here.